_posts/2017-09-22-redis-2-to-redis-3.md

```diff
@@ -0,0 +1,48 @@
+
+All that remained was the redis instances that served our php code (well one of them anyway, the other turned out to be unnessacary in the end) and we hit a thorn, actually a couple, in the road. First, and most pressing, was finding a redis cluster aware driver that we could use in php, and the version of php we were using. This proved to be something that was doable, because we had upgraded to a recent version of php. Alas the driver we chose did not like using redis auth, something we had decided to use with redis cluster as an extra security step (I told you there would be more on that security thing). As we were replacing each legacy redis instance with it's own redis cluster the fix seemed straight forward, turn redis auth off, and all would be well with the world. However this did not prove to be true, for some reason having done this the redis cluster wouldn't accept connections from the web servers. A new security feature introduced by redis in version 3 called protected mode would stop redis listening to connections from external IP addresses when redis was bound to any interface, and no redis auth password was configured. This proved reasonably easy to fix, but caught us off guard.
+
+### And now?
+
+So this is where we find ourselves. We have migrated off some of our legacy redis instances, and are migrating off the rest. We have, by way of doing this solved some of our technical debt, and improved our platform's stability. With redis cluster we can also scale out the in memory databases as well as scale them up. Redis is single threaded, so just throwing more memory at a single instance is only ever going to allow so much growth, and we are already nipping at the heals of that limit. We're expecting an improved perfromance from the new cluster, as well as it giving us more options for expansion and load balancing.
+
+### What about the Future?
+
+So we have solved some technical debt, and made our services easier to support, and more stable. That doesn't mean the job is done, indeed Redis 4 appears to have some features that we may want to look into. And Redis isn't the only software we use. We will continue to work to improve the platform, and reduce the time spent dealing with technical debt, but as our customer base expands, and we strive to offer ever richer services, we are always going to end up with things that need improving. The next challange is likely to be related to scaling from more than 20,000 logins within a minute to more than 40,000, and even beyond.
```

Comment:
s/heals/heels/

I'd also remove the 'within' from the 20,000 logins within a minute to more...